plugins/scanners/fossid/src/test/kotlin/TestUtils.kt

```diff
@@ -737,3 +738,60 @@ internal fun FossId.scan(
             snippetChoices = snippetChoices
         )
     )
+
+/**
+ * Mock all the different functions called during an upload archive workflow for a new scan [scanCode].
+ * If [deltaScans] is enabled, it also mocks the checks for existing scans.
+ * If [existingScans] is not empty, it also mocks the checks for the contained scans status.
+ */
+@Suppress("LongParameterList")
+internal fun FossIdServiceWithVersion.expectUploadArchiveWorkflow(
+    projectCode: String,
+    scanCode: String,
+    vcsInfo: VcsInfo,
+    projectRevision: String,
+    existingScans: List<Scan> = emptyList(),
+    deltaScans: Boolean
+): FossIdServiceWithVersion =
+    apply {
+        expectProjectRequest(projectCode)
+        expectListScans(projectCode, existingScans)
+        // TODO : this should be done only if deltaScans is true, but currently the tests with delta scans disabled
+        //  still rely on existing scans.
```

Comment:
Wrong style, should be "TODO: This" (no space before the colon, start capitalized) and the next line start should be aligned with the previous line start (to the actual text, not to the "TODO:" statement).